src/core/src/FabricPrepareRenderResources.cpp

```diff
@@ -404,7 +407,33 @@ FabricPrepareRenderResources::prepareInLoadThread(
     // but at least we have an upper bound. Unused texture slots are initialized with a 1x1 transparent pixel so
     // blending still works.
     const auto overlapsImagery = tileLoadResult.rasterOverlayDetails.has_value();
-    const auto imageryLayerCount = overlapsImagery ? _tileset->getImageryLayerCount() : 0;
+    ImageryLayersInfo imageryLayersInfo;
+    auto imageryLayerCount = overlapsImagery ? _tileset->getImageryLayerCount() : 0;
+    if (imageryLayerCount > 0) {
+        auto stage = Context::instance().getStage();
+        for (uint64_t i = 0; i < imageryLayerCount; i++) {
+            auto imageryLayerPath = _tileset->getImageryLayerPath(static_cast<int>(i));
+            auto prim = stage->GetPrimAtPath(imageryLayerPath);
+
+            if (prim.IsA<pxr::CesiumIonImagery>()) {
+                imageryLayersInfo.overlayTypes.emplace_back(OverlayType::ION);
+            } else if (prim.IsA<pxr::CesiumPolygonImagery>()) {
+                imageryLayersInfo.overlayTypes.emplace_back(OverlayType::POLYGON);
+            } else {
+                Context::instance().getLogger()->error(
+                    "Attempting to prepare resources for unknown imagery layer type");
+            }
+
+            auto imageryLayer = UsdUtil::getCesiumImagery(imageryLayerPath);
+            pxr::TfToken overlayRenderPipe;
+            imageryLayer.GetOverlayRenderPipeAttr().Get<pxr::TfToken>(&overlayRenderPipe);
+            if (overlayRenderPipe == pxr::CesiumTokens->overlay) {
+                imageryLayersInfo.overlayRenderPipes.emplace_back(OverlayRenderPipe::OVERLAY);
+            } else if (overlayRenderPipe == pxr::CesiumTokens->clip) {
+                imageryLayersInfo.overlayRenderPipes.emplace_back(OverlayRenderPipe::CLIPPING);
+            }
```

Comment:
This could be a getter in `OmniImagery`, similar to `OmniTileset::getSourceType`.

One of the themes of my cleanup branch is to move most of the USD API usage into `Omni` classes.

Reply:
Yes. I like that placement much better. I had to change `OmniImagery` from and abstract base class to a base class, but that makes sense to me for querying common properties of raster overlays.